pkgs/development/compilers/julia/generic.nix

```diff
@@ -3,6 +3,10 @@
 , patches
 }:
 
+let
+  pkgs = import <nixpkgs> {};
```

Comment:
You don't want to do this -- you should add the attributes you need to the imports at the top. It looks like you already did this with `xcbuild`.

Reply:
How should I call `runCommands` in this case?

Reply:
You just add `runCommand` as one of the inputs and then it will be available in scope.

But I definitely think you should try putting the `darwin` input into the `buildInputs` so the Makefile doesn't need patching at all, and also avoid the IFD @NickCao mentioned.

Reply:
I tried putting it into `buildInputs` but got:
```nix
       error: Dependency is not of a valid type: element 2 of buildInputs for julia
```